src/dev/flang/fe/Module.java

```diff
@@ -269,6 +268,62 @@ else if (existing == f && !existing.generics().equals(f.generics()) ||
   }
 
 
+  /**
+   * Is type defined by feature `af` visible in file `usedIn`?
+   * If `af` does not define a type, result is false.
+   *
+   * @param usedIn
+   * @param af
+   * @return
+   */
+  protected boolean typeVisible(SourceFile usedIn, AbstractFeature af)
+  {
+    var m = (af instanceof LibraryFeature lf) ? lf._libModule : this;
```

Comment:
Avoid `instanceof`, instead please add a method `libModule()` or similar to `AbstractFeature' that returns `null` or the source module and that is redefined in `LibraryFeature`.

Reply:
but then ast needs to dependend on fe?